### PR TITLE
fix(ally): add alt attribute for components using Icon

### PR DESCRIPTION
**Pull request by lilyclyc**

### CODE REVIEW

**Comment by lilyclyc** [comment]
@chrisdhanaraj -- Please help to review it, thank you.

File: components/Checkbox.js [hunk 1]
```diff
@@ -41,6 +41,7 @@ const Checkbox = ({ className, id, labelText, onChange, ...other }) => {
           <Icon
             className="bx--checkbox-checkmark"
             description="checkmark"
+            alt="checkmark"
```
Comment (chrisdhanaraj) [inline components/Checkbox.js]: for these `alt` tags and below, could you actually move them to a `defaultProp` and then reference that prop here? It'll allow for people to pass in different internationalized strings if need be

Comment (hellobrian) [inline components/Checkbox.js]: It was pointed out in https://github.com/carbon-design-system/carbon-components-react/issues/87#issuecomment-314853593 that we need to add `alt` attributes to all of our `Icon` components any way. 

So in `Icon`, couldn't we just add `alt` as a prop and as a defaultProp?

Comment (shyfruan) [inline components/Checkbox.js]: Have created a prop "iconDescription" to replace the hard coded value "checkmark"

File: components/Checkbox.js [hunk 2]
```diff
@@ -40,8 +42,8 @@ const Checkbox = ({ className, id, labelText, onChange, ...other }) => {
         <span className="bx--checkbox-appearance">
           <Icon
             className="bx--checkbox-checkmark"
-            description="checkmark"
-            alt="checkmark"
+            description={iconDescription}
+            alt={iconDescription}
```
Comment (chrisdhanaraj) [inline components/Checkbox.js]: could you actually change this into two different props

```
descriptionText
altText
```

with both defaults being the same in this scenario?

Comment (shyfruan) [inline components/Checkbox.js]: Have created another prop iconAlt

**Comment by shyfruan** [comment]
@chrisdhanaraj conflict has been resolved and any more comments?

**Comment by chrisdhanaraj** [comment]
👍  good by me

**Pull request closed without merging**